Node Upgrade

SabreCat opened the pull request:
[//]: # (Note: See https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Attempts to upgrade NodeJS to latest version. Version 14 which Habitica currently uses is no longer supported.

SabreCat committed:
chore(node): upgrade to node@latest

SabreCat committed:
fix(CI): change target Node version

SabreCat committed:
fix(CI): try normal npm install

SabreCat committed:
fix(ci): update checkout and node actions

SabreCat committed:
fix(CI): try installing libkrb5-dev

SabreCat committed:
fix(CI): passwordless sudo

SabreCat committed:
fix(lint): lint autofix

SabreCat committed:
fix(CI): add libkrb5-dev to all tasks

SabreCat committed:
fix(build): package changes for prod

SabreCat committed:
fix(packages): update client lint some more

negue committed:
fix svg loading - onto the next issues.

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json
#	website/client/package-lock.json
#	website/client/package.json
#	website/client/src/components/static/faq.vue

negue committed:
getting the failed tests to run again

negue committed:
fix lint

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

negue committed:
fix lint

negue commented on test/api/v3/integration/challenges/POST-challenge_flag.test.js:
```diff
@@ -41,11 +41,11 @@ describe('POST /challenges/:challengeId/flag', () => {
 
   it('flags a challenge with a higher count when from an admin', async () => {
-    await user.update({ 'contributor.admin': true });
+    const admin = await generateUser({ 'contributor.admin': true });
```
TODO for me - revert this again and see if its working again

might have something to do with the `domain-middleware` that the CI's stopped for it

negue replied:
will try that again tomorrow

rafucuk commented:
What's left with this PR

negue commented on .github/workflows/test.yml:
```diff
@@ -228,15 +237,16 @@ jobs:
+      uses: actions/setup-node@v4
       with:
         node-version: ${{ matrix.node-version }}
+    - run: sudo apt-get -y install libkrb5-dev
```
@SabreCat which program needed this dependency?

negue commented:
@SabreCat created #15040 to offload the storybook changes of this PR - could you take a look at maybe just merge - then this one is a bit less cluttered

negue committed:
revert test and middleware changes, only add domainMiddleware when cluster is enabled

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json

negue committed:
remove .only

negue committed:
fix lints

negue committed:
fix lint

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json
#	website/client/package-lock.json

negue committed:
Merge remote-tracking branch 'origin/sabrecat/node-latest' into sabrecat/node-latest

negue committed:
set node version to 20

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json
#	test/.eslintrc.js
#	website/client/package-lock.json
#	website/client/package.json
#	website/client/src/store/actions/user.js
#	website/client/src/store/index.js

negue committed:
fix conflicts

negue committed:
remove svg width/height + fix lints

negue committed:
fix lints

SabreCat committed:
fix(tests): improve property assignments

SabreCat committed:
fix(svg): use working Apple icon

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json
#	test/.eslintrc.js
#	website/client/package-lock.json
#	website/client/package.json
#	website/client/src/store/actions/user.js
#	website/client/src/store/index.js

negue committed:
merge develop into node-latest

negue committed:
lint

negue committed:
fixed many svgs to scale normally again (width/height need to be removed, viewbox stays) + SVGs in URLS wont be inlined

negue committed:
Merge remote-tracking branch 'origin/sabrecat/node-latest' into sabrecat/node-latest

# Conflicts:
#	.eslintrc.js
#	package-lock.json
#	test/.eslintrc.js
#	website/client/src/assets/svg/experience.svg
#	website/client/src/assets/svg/gem-red.svg
#	website/client/src/assets/svg/gem.svg
#	website/client/src/assets/svg/hourglass-red.svg
#	website/client/src/assets/svg/hourglass.svg
#	website/client/src/assets/svg/streak.svg
#	website/client/src/assets/svg/warrior.svg
#	website/server/controllers/api-v3/chat.js

SabreCat committed:
fix(lint): whitespace

SabreCat committed:
Merge branch 'release' into sabrecat/node-latest

SabreCat committed:
fix(equip): remove hardcoded SVG fill color

SabreCat committed:
fix(imports): don't use default
Also fixes an issue where process.env might be referenced before it's ready on the client

SabreCat committed:
fix(lint): shush noisy tag order

SabreCat committed:
Merge branch 'release' into sabrecat/node-latest

SabreCat committed:
chore(npm): update package locks

negue committed:
DEFINE process.env.vars + refactor externalLinks

negue committed:
Merge remote-tracking branch 'origin/develop' into sabrecat/node-latest

# Conflicts:
#	package-lock.json

negue committed:
change habitica.com in trusted domains to https://habitica.com

SabreCat committed:
fix(lint): eof newlines

SabreCat committed:
Merge branch 'release' into sabrecat/node-latest

SabreCat committed:
fix(CI): node 20 not 21
